grails-app/controllers/com/unifina/controller/data/StreamController.groovy

```diff
@@ -165,7 +165,7 @@ class StreamController {
 					response.status = 500
 					render([success: false, redirect: createLink(action: 'confirm', params: [id: params.id, file: temp.getCanonicalPath()])] as JSON)
 				} else {
-					Map updatedConfig = streamService.importCsv(csv, stream)
+					Map updatedConfig = streamService.importCsv(csv, stream, user.username)
```

Comment:
This might need some more discussion/thought, but my first instinct is that we should probably use the `user.id` as the id for non-Ethereum users to avoid exposing the user's email address to subscribers.

The logic for what to use as the `producerId` could be abstracted by `SecUser`, as now you notice that you'll have to change quite many places to change the logic. Something like:

```
String getProducerId() {
    // If Ethereum user return username (=address)
    // Else return id.toString()
}
```